### DISCUSSION TITLE
User/dev mode

### DISCUSSION THREAD

**Discussion by 3djc** [gui-dos/DiaBLE#5] [General]
Some of the screens /fields are really more dev oriented, so I wonder if it makes sense to create a dev mode toggle in settings that hides those for end users

**Comment by gui-dos** [comment]
Currently I am defining three user levels, “Basic”, “Devel” and “Test”:

https://github.com/gui-dos/DiaBLE/blob/24c8e1485dbc43b20a8418210b0630abf925df2c/DiaBLE/App.swift#L141-L144

They wouldn’t necessarily be tied to the `LogLevel` enum and they are currently bound to the middle button of the console sidebar , comfortably at the thumb reach.
![Trident Modes](https://github.com/gui-dos/DiaBLE/assets/7220550/ef1543fe-d345-4b6f-931e-17736222666c)
Switching to "Test" mode back and forth can avoid sending the "unlock payload" to the sensor, and I found that interfering with Trident by sending random payloads can force it to repeat the same sequence as the security commands `01`, `02`, `03  ...` sent during the sensor activation so as to sniff and analyze them  ;-)

Since I am targeting the #wearenotwaiting geek community, I think I'll keep this douBLE approach merged: that's also why I chose a bi-color logo B-)

It is also my intention to rename the essential text-mode central tab from "Console" to "Copilot", as is so trending today, but it is not so smart yet... :-)
